src/org/traccar/helper/Parser.java

```diff
@@ -165,4 +167,122 @@ public double nextCoordinate() {
         return nextCoordinate(CoordinateFormat.DEG_MIN_HEM);
     }
 
+    public enum DateTimeFormat {
+        HMS,         // HHMMSS
+        SMH,         // SSMMHH
+
+        HMS_YMD,     // HHMMSSYYYYMMDD or HHMMSSYYMMDD
+        HMS_DMY,     // HHMMSSDDMMYYYY or HHMMSSDDMMYY
+        SMH_YMD,     // SSMMHHYYYYMMDD or SSMMHHYYMMDD
+        SMH_DMY,     // SSMMHHDDMMYYYY or SSMMHHDDMMYY
+
+        DMY_HMS,     // DDMMYYYYHHMMSS or DDMMYYHHMMSS
+        DMY_HMSms,   // DDMMYYYYHHMMSS.sss or DDMMYYHHMMSS.sss
```

Comment:
It should all be uppercase. I suggest put double-S when there are milliseconds. Milliseconds are basically decimals of the seconds.